### PR TITLE
feat(ui-color-picker,ui-color-utils): add callback for contrast validation information and export validation methods

### PR DESCRIPTION
**Pull request by HerrTopi**
TEST_PLAN:
check if contrast calculation works as before for ColorContrast.
Test if you pass the onContrastChange callback, to ColorContrast or ColorPicker, it really gets called when the color changes.

Check if preciously internally used methods work as before in color utils

Check if validationLevel prop does what it says

Closes: INSTUI-4360

### CODE REVIEW

**Review by matyasf — CHANGES_REQUESTED** [review]
nice work, just some small documentation fixes from me

File: packages/ui-color-utils/src/overlayColors.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * @param {RGBAType} c2
+ * @returns {RGBAType} color as rgb string
+ */
+const overlayColors = (c1: RGBAType, c2: RGBAType): RGBAType => {
+  const alpha = 1 - (1 - c1.a) * (1 - c2.a)
+  return {
+    r: (c2.r * c2.a) / alpha + (c1.r * c1.a * (1 - c2.a)) / alpha,
+    g: (c2.g * c2.a) / alpha + (c1.g * c1.a * (1 - c2.a)) / alpha,
+    b: (c2.b * c2.a) / alpha + (c1.b * c1.a * (1 - c2.a)) / alpha,
+    a: 1
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-color-utils/src/overlayColors.ts:53]: Done

File: packages/ui-color-utils/src/validateContrast.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+/**
+ * ---
+ * category: utilities
+ * ---
+ * Decides if the given contrast is sufficient for different text sizes and situations.
+ *  for normalText, at least 4.5:1 contrast is required
+ *  for largeText, at least 3:1 contrast is required
+ *  for graphicsText, at least 3:1 contrast is required
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-color-utils/src/validateContrast.ts:45]: Done

File: packages/ui-color-picker/src/ColorContrast/props.ts [hunk 3]
```diff
@@ -83,6 +83,27 @@ type ColorContrastOwnProps = {
...
+   * Triggers a callback whenever the contrast changes, due to a changing color input.
+   * Communicates the contrast and the success/fail state of the contrast, depending on
+   * the situation:
+   *
+   * isValidNormalText true if at least 4.5:1
+   *
+   * isValidLargeText true if at least 3:1
+   *
+   * isValidGraphicsText true if at least 3:1
+   */
```
[the remark this answers is not in the document]
Comment (HerrTopi) [reply packages/ui-color-picker/src/ColorContrast/props.ts:96]: Done

**Review by balzss — APPROVED** [review]

**Review by matyasf — APPROVED** [review]

**Pull request merged**